sdk/scheduler/src/main/java/com/mesosphere/sdk/curator/CuratorStateStore.java

```diff
@@ -284,7 +284,7 @@ public void clearTask(String taskName) throws StateStoreException {
         try {
             byte[] bytes = curator.get(path);
             if (bytes.length > 0) {
-                return Optional.of(Protos.TaskInfo.parseFrom(bytes));
+                return Optional.of(CommonTaskUtils.unpackTaskInfo(Protos.TaskInfo.parseFrom(bytes)));
```

Comment:
StateStore is storing packed `TaskInfo`? Shouldn't we only pack it where we have to, e.g. where we send it to Mesos?